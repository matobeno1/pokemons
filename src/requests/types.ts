export type FetchPokemonsRequestResponseType = Readonly<{
    results: Array<{
        name: string,
        url: string
    }>
}>;
