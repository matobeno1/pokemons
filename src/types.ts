export type Pokemon = {
    id: string;
    name: string;
    weight: number
    height: number,
};
