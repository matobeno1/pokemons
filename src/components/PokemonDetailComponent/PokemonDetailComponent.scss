@import "styles/mixins";

.pokemon-detail-component {
  height: 100vh;
  position: relative;


  &__backdrop {
    @include backdrop;
    @include background-filter;
    background-color: var(--background-color);
  }

  &__heading {
    color: var(--color-text)
  }

  &__image {
    display: flex;
    justify-content: center;
  }

  &__card {
    height: 100%;
    background-color: #ffffff;
    border-radius: 0 var(--spacing-5);
  }
}
