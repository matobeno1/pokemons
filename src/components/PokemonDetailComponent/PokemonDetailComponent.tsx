import { FunctionComponent } from "react";

export type PokemonDetailComponentProps = {

};

export const PokemonDetailComponent: FunctionComponent<PokemonDetailComponentProps> = ({

}) => (
    <div>
        detail
    </div>
);
