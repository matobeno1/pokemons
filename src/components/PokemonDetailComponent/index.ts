export { PokemonDetailComponent } from "./PokemonDetailComponent";
export type { PokemonDetailComponentProps } from "./PokemonDetailComponent";
