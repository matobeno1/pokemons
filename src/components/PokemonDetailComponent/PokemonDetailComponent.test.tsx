import { act, create } from "react-test-renderer";
import { PokemonDetailComponent } from "./PokemonDetailComponent";
import { Pokemon } from "../../types";

describe(PokemonDetailComponent, () => {
    it("snapshots", async () => {
        let root: ReturnType<typeof create> | undefined;
        const pokemon: Pokemon = {
            id: "1",
            name: "bulbasaur",
            weight: 50,
            height: 50,
            color: {
                name: "green"
            }
        };
        act(() => {
            root = create(<PokemonDetailComponent pokemon={pokemon} />);
        });
        expect(root?.toJSON()).toMatchSnapshot();
    });
});
