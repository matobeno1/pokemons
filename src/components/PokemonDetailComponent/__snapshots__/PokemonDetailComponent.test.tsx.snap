// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonDetailComponent snapshots 1`] = `
<div
  className="pokemon-detail-component"
  style={
    Object {
      "--background-color": "green",
    }
  }
>
  <div
    className="pokemon-detail-component__backdrop"
  />
  <div
    className="pokemon-detail-component__card"
  >
    <div
      className="pokemon-detail-component__image"
    >
      <img
        alt="bulbasaur"
        draggable={false}
        layout="fill"
        src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/1.svg"
      />
    </div>
    <div
      className="pokemon-detail-component__description"
    >
      <div
        className="pokemon-detail-component__heading"
      >
        <h1>
          bulbasaur
           - #
          001
        </h1>
      </div>
      <div>
        <h2>
          Basic information
        </h2>
        <b>
          Weight: 
        </b>
        <span>
          50
        </span>
        <br />
        <b>
          Height: 
        </b>
        <span>
          50
        </span>
      </div>
    </div>
  </div>
</div>
`;
